File: subprojects/java-compatibility/src/test/java/kotools/types/experimental/AnyIntFactoryTest.java
/*
 * Copyright 2023 Kotools S.A.S.U.
 * Use of this source code is governed by the MIT license.
 */

package kotools.types.experimental;

import kotools.types.number.AnyInt;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.Random;

public class AnyIntFactoryTest {
    @Test
    public void createShouldPassWithInt() {
        final int expected = new Random()
                .nextInt();
        final AnyInt number = AnyIntFactory.create(expected);
        final int actual = number.toInt();
        Assertions.assertEquals(expected, actual);
    }
}

File: subprojects/library/src/commonTest/kotlin/kotools/types/experimental/AnyIntFactoryTest.kt
package kotools.types.experimental

import kotools.types.number.AnyInt
import kotools.types.number.ZeroInt
import kotools.types.number.toStrictlyNegativeInt
import kotools.types.number.toStrictlyPositiveInt
import org.kotools.types.ExperimentalKotoolsTypesApi
import kotlin.random.Random
import kotlin.test.Test
import kotlin.test.assertEquals

class AnyIntFactoryTest {
    @ExperimentalKotoolsTypesApi
    @Test
    fun constructor_like_function_should_pass_with_Int() {
        val value: Int = Random.nextInt()
        val actual = AnyInt(value)
        val expected: AnyInt = when {
            value > 0 -> value.toStrictlyPositiveInt().getOrThrow()
            value < 0 -> value.toStrictlyNegativeInt().getOrThrow()
            else -> ZeroInt
        }
        assertEquals(expected, actual)
    }
}
